src/gui/tray/UnifiedSearchResultsListModel.cpp

```diff
@@ -0,0 +1,675 @@
+        return QStringLiteral("qrc:///client/theme/black/edit.svg");
+    } else if (providerId.contains(QStringLiteral("deck"), Qt::CaseInsensitive)) {
+        return QStringLiteral("qrc:///client/theme/black/deck.svg");
+    } else if (providerId.contains(QStringLiteral("calendar"), Qt::CaseInsensitive)) {
+        return QStringLiteral("qrc:///client/theme/black/calendar.svg");
+    } else if (providerId.contains(QStringLiteral("mail"), Qt::CaseInsensitive)) {
+        return QStringLiteral("qrc:///client/theme/black/email.svg");
+    } else if (providerId.contains(QStringLiteral("comment"), Qt::CaseInsensitive)) {
+        return QStringLiteral("qrc:///client/theme/black/comment.svg");
+    }
+
+    return QStringLiteral("qrc:///client/theme/change.svg");
+}
+
+static QString iconUrlForDefaultIconName(const QString &defaultIconName)
+{
+    const QUrl urlForIcon(defaultIconName);
+
+    if (urlForIcon.isValid() && !urlForIcon.scheme().isEmpty()) {
+        return defaultIconName;
+    }
+    
+    if (defaultIconName.startsWith(QStringLiteral("icon-"))) {
+        const auto parts = defaultIconName.split(QLatin1Char('-'));
+
+        if (parts.size() > 1) {
+            const QString iconFilePath = QStringLiteral(":/client/theme/") + parts[1] + QStringLiteral(".svg");
+
+            if (QFile(iconFilePath).exists()) {
+                return iconFilePath;
+            }
+
+            const QString blackIconFilePath = QStringLiteral(":/client/theme/black/") + parts[1] + QStringLiteral(".svg");
+
+            if (QFile(blackIconFilePath).exists()) {
+                return blackIconFilePath;
+            }
+        }
+
+        if (defaultIconName.contains(QStringLiteral("message"), Qt::CaseInsensitive)
+            || defaultIconName.contains(QStringLiteral("talk"), Qt::CaseInsensitive)) {
+            return QStringLiteral(":/client/theme/black/wizard-talk.svg");
+        } else if (defaultIconName.contains(QStringLiteral("folder"), Qt::CaseInsensitive)) {
+            return QStringLiteral(":/client/theme/black/folder.svg");
+        } else if (defaultIconName.contains(QStringLiteral("deck"), Qt::CaseInsensitive)) {
+            return QStringLiteral(":/client/theme/black/deck.svg");
+        } else if (defaultIconName.contains(QStringLiteral("contacts"), Qt::CaseInsensitive)) {
+            return QStringLiteral(":/client/theme/black/wizard-groupware.svg");
+        } else if (defaultIconName.contains(QStringLiteral("calendar"), Qt::CaseInsensitive)) {
+            return QStringLiteral(":/client/theme/black/calendar.svg");
+        } else if (defaultIconName.contains(QStringLiteral("mail"), Qt::CaseInsensitive)) {
+            return QStringLiteral(":/client/theme/black/email.svg");
+        }
+    }
+
+    return QStringLiteral(":/client/theme/change.svg");
+}
+
+static QString iconsFromThumbnailAndFallbackIcon(QString thumbnailUrl, QString fallackIcon, QUrl serverUrl)
```

Comment:
same, and why are the args not `const&`

Reply:
@FlexW. Non-const since I am modifying them and found it convenient to just reuse the args instead of creating redundant copies. Any other thoughts?